src/app/pages/applications/dialogs/pod-select/pod-select-dialog.component.spec.ts

```diff
@@ -97,5 +100,65 @@ describe('PodSelectDialogComponent', () => {
       expect(spectator.inject(Router).navigate).toHaveBeenCalledWith(['/apps/1/logs/', 'app_name', 'pod2', 'container21', '500']);
       expect(spectator.inject(MatDialogRef).close).toHaveBeenCalled();
     });
+
+    it('warning dialog should be displayed if there are no pods', () => {
+      const ws = spectator.inject(MockWebsocketService);
+      ws.mockCall('chart.release.pod_console_choices', {});
+      spectator.component.ngOnInit();
+      spectator.detectChanges();
+      const dialogContent = spectator.query('.mat-dialog-content ');
+      expect(dialogContent).toHaveText('At least one pool must be available to use apps');
+    });
+  });
+
+  describe('dialog type is PodLogs', () => {
+    const mockCustomSubmit = jest.fn();
+    const objSpy = {
+      type: PodSelectDialogType.PodLogs,
+      customSubmit: (dialog: PodSelectDialogComponent) => mockCustomSubmit(dialog),
+      afterDialogInit: () => {},
+    };
+    const apps = [{ label: 'nextcloud-app1', value: 'nextcloud-app1' }, { label: 'nextcloud-app2', value: 'nextcloud-app2' }];
+    const pods = [{ label: 'nextcloud-pod1', value: 'nextcloud-pod1' }, { label: 'nextcloud-pod2', value: 'nextcloud-pod2' }];
+    const containers = [{ label: 'nextcloud-container1', value: 'nextcloud-container1' }, { label: 'nextcloud-container2', value: 'nextcloud-container2' }];
+    const fillForm = async (): Promise<any> => {
+      spectator.component.apps$ = of(apps);
```

Comment:
I would not set these properties directly. They are not really meant to be interacted with outside of the component.
I would inject `MockWebsocketService` and re-mock necessary calls.

Reply:
Dialog for PodLogsComponent extracted into separate component.